src/main/java/hlf/java/rest/client/service/TransactionFulfillment.java

```diff
@@ -135,4 +138,27 @@ ResponseEntity<EventAPIResponseModel> readTransactionEventByBlockNumber(
       String transactionId,
       String chaincode,
       String eventType);
+
+  /**
+   * Handles the ledger write operation for scenarios where incoming request contains multiple PDC
+   * or Public data details.
+   *
+   * <p>Returns Network response for Successful write operation, wrapped within ResponseEntity
+   * Model.
+   *
+   * <p>For failures, response would be wrapped within an Exception, that can potentially be of kind
+   * {@link GatewayRuntimeException} or {@link ContractException} or {@link InterruptedException}
+   * depending on the nature of failure.
+   *
+   * @param networkName String channel name
+   * @param contractName String chaincode name
+   * @param transactionFunctionName String function name in chaincode
+   * @param multiDataTransactionPayload String The arguments as Json Payload
+   * @return responseEntity ResponseEntity Transaction Response
+   */
+  ResponseEntity<ClientResponseModel> writeMultiDataTransactionToLedger(
+      String networkName,
```

Comment:
[nitpick] can we have uniformity across all methods, at controller level we are calling this field as `channelName`

Reply:
Agree, renaming to channelName.